plexapi/library.py

```diff
@@ -436,21 +441,80 @@ def reload(self, key=None):
         return self._server.library.section(self.title)
 
     def edit(self, agent=None, **kwargs):
-        """ Edit a library (Note: agent is required). See :class:`~plexapi.library.Library` for example usage.
+        """ Edit a library. See :class:`~plexapi.library.Library` for example usage.
 
             Parameters:
+                agent (str, optional): The library agent.
                 kwargs (dict): Dict of settings to edit.
+
+            Example:
+
+                .. code-block:: python
+                LibrarySection.edit(**{'location': ['/path/1', 'path/2', '/path/3']})
         """
         if not agent:
             agent = self.agent
-        part = '/library/sections/%s?agent=%s&%s' % (self.key, agent, urlencode(kwargs))
+
+        if kwargs['location']:
```

Comment:
```py
if 'location' in kwargs:
    # Assume paths have already been validated in addLocations and removeLocations
    locations = [('location', path) for path in kwargs.pop('location')]

params = list(kwargs.items()) + locations
```

Reply:
This assumes that a user would use the convenience method over `edit` for changing locations. Should the validation happen here instead then?